### ISSUE TITLE
[Screen reader] Rename 'Remove constraint ..' to 'Remove Filter ..'

### ISSUE THREAD

**Issue by christinach** [pulibrary/orangelight#4891]
### Steps to recreate
1. Open Voice over. 
2. Go to https://catalog.princeton.edu/  
3. Do a search. For example search for 'plasticity'.
4. Let the screen reader read the page. It reads 'Remove constraint plasticity' 

### URL
https://catalog.princeton.edu/catalog?search_field=all_fields&q=plasticity

### Issues
1. Confusing for the tester. 
2. Very specific to blacklight users.
3. Not intuitive.

### Severity
High

### WCAG Conformance level (A, AA, AAA)
N/A

### User impact
1. Confusing for the tester. 
2. Very specific to blacklight users
3. Not clear to the user what they have to do.

### Recommended fix
Rename 'Remove constraint' to 'Remove filter'
```
<span class="sr-only visually-hidden">Remove constraint plasticity</span>
```

### Acceptance criteria
- [ ] Screen reader reads `Remove filter <filter term>`
- [ ] I can successfully use this feature with Voice Over in Safari.
- [ ] I have checked this feature using one of the following browser extensions and confirmed that it does not add any level A or level AA issues:
    * axe dev tools
    * Lighthouse
    * Wave
- [ ] ...

### Notes
Related to https://github.com/pulibrary/orangelight/issues/4888
